EIPS/EIP-1776.md

```diff
@@ -0,0 +1,187 @@
+---
+eip: 1776
+title: ERC-721 Royalty Standard
+author: Zach Burks (@vexycats)
+discussions-to: https://github.com/ethereum/EIPs/issues/2907
+status: First Draft
+type: Standards Track
+category: ERC
+created: 2020-09-15
+requires: 721
+---
+
+## Simple Summary
+
+A standardized way to handle royalty payments for ERC-721 tokens, including publically viewable information and notification of payment event.
+
+## Abstract
+
+Many of the largest ERC-721 marketplaces have implemented a form of royalties that is not compatible across the board and therefore worthless if the NFT is sold on another marketplace, defeating the purpose of any royalty system. This standard is a proposed way to minimally implement royalties that can be accepted across any type of NFT marketplace smart contracts. This standard allows for a royalty standard to be accepted on all marketplaces - leaving the funds transfer up to the marketplace itself, and only providing a means to fetch the royalty amounts and a event to be fired off when transfer has happened.  
+
+## Motivation
+
+This extension provides even more flexibilty to the [ERC-721 specification](./eip-721.md). It is possible to set a rooyalty amount that can be paid to the creator on any marketplace that implements this ERC. If a marketplace chooses not to implement this ERC then of course no funds are paid for secondary sales. But seeing as how most NFT marketplace have developed some sort of royalty system themselves - and all of them are singular and only work on their own contracts - there needs to be an accepted standard way of providing royalties - if the creator so chooses to set royalties on their NFTs.
+
+Many decentralized marketplaces use different means to facilitate the trade of NFTs - therefore this ERC does not propose a method of transfering funds - but it simply provides a means to return the royalty amounts, and a function to be called once the transfer has been completed. Allowing each marketplace to easily integrate the standard in whatever flow they currently use for their marketplace.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL
+NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and
+"OPTIONAL" in this document are to be interpreted as described in
+RFC 2119.
+
+**ERC-721 compliant contracts MAY implement this ERC for royalties to provide a standard method of accepting royalty payments and recieving royalty information**
+
+The `_RoyaltyAmount` **MUST** be calculated as a percentage - such as "5" - for 5%. This is **REQUIRED** to maintain uniformity across the standard. 
```

Comment:
I choose fixed point with scaling to 100 as all the other marketplaces with royalties do not let you go below 1%, with the average royalty percentage being 1-10%. 

I'm not opposed to making it go lower. I'm not sure which would be best, a scaling factor of 10,000, or scaled to 10^18. 

Most users would not need/select a percentage of 0.00005% though - as an NFT purchase is a onetime transaction and not like a general fee on all transactions on something like say uniswap. 

Continuing to go through these and replying. Thank you for the feedback Michah!

Reply:
I used 10,000 as the scaling factor as I don't see any reason someone would want to go below that for an NFT royalty fee. 

Im unsure if its a good move, as for adopting the standard a marketplace really needs to make sure its math is correct and it uses the 10000 scale factor, otherwise it might  have a bad time..... I'd like an easier solution where it requires less math/attention to detail for implementation, but alas I don't know of what would be better

Reply:
The advantage of `10^18` is that it is *very* standard across Ethereum, so it is more likely that tools will "just work" with that then with some custom scaling factor.  The advantage of `100` is that it is easy to just describe the value as a "percentage".  Personally, I would just do `10^18` I think because I believe that 100 is way too limiting (I can imagine people wanting sub percentage royalties) and I think standardization is valuable.  You aren't really saving anything by using 10,000 as all numbers in Solidity use the same number of bytes of storage (unless you are bit packing, which I don't think you are or should).